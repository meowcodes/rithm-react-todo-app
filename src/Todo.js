import React, { Component } from 'react';

class Todo extends Component {

    render() {
        return (
            <li>
                <p>{ this.props.text }</p>
                <button onClick={ this.props.notifyDelete }>X</button>
                <button onClick={ this.props.notifyDelete }>X</button>
            </li>)
    }
}

export default Todo;