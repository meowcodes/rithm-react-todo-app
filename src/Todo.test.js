import React from "react";
import { mount } from "enzyme";
import Todo from "./Todo";
import toJson from "enzyme-to-json";

let wrapper;

beforeEach(function () {
    wrapper = mount(<Todo text="need to test" />);
})


it("renders without crashing", function () {
    mount(<Todo />);
});

it("matches snapshot", function () {
    let serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
});

it("renders with correct text", function () {
    let p_elem = wrapper.find("p");
    expect(p_elem.text()).toBe("need to test");
})

it("renders a button", function () {
    expect(wrapper.exists("button")).toEqual(true);
});