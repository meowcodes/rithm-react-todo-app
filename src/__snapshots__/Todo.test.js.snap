// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<Todo
  text="need to test"
>
  <li>
    <p>
      need to test
    </p>
    <button>
      X
    </button>
  </li>
</Todo>
`;
